diff --git a/gcc-4.9/libgfortran/mk-kinds-h.sh b/gcc-4.9/libgfortran/mk-kinds-h.sh
index 98a95ab..7cbe7ed 100755
--- a/gcc-4.9/libgfortran/mk-kinds-h.sh
+++ b/gcc-4.9/libgfortran/mk-kinds-h.sh
@@ -62,7 +62,7 @@ for k in $possible_real_kinds; do
       4) ctype="float" ; cplxtype="complex float" ; suffix="f" ;;
       8) ctype="double" ; cplxtype="complex double" ; suffix="" ;;
       10) ctype="long double" ; cplxtype="complex long double" ; suffix="l" ;;
-      16) if [ $long_double_kind -eq 10 ]; then
+      16) if [ $long_double_kind -ne 16 ]; then
            ctype="__float128"
            cplxtype="_Complex float __attribute__((mode(TC)))"
            suffix="q"
